hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/om/CompactionLogDagPrinter.java

```diff
@@ -0,0 +1,220 @@
+import com.google.protobuf.InvalidProtocolBufferException;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.concurrent.Callable;
+import java.util.concurrent.ConcurrentHashMap;
+import org.apache.hadoop.hdds.protocol.proto.HddsProtos;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedRocksDB;
+import org.apache.hadoop.hdds.utils.db.managed.ManagedRocksIterator;
+import org.apache.hadoop.ozone.debug.RocksDBUtils;
+import org.apache.ozone.compaction.log.CompactionFileInfo;
+import org.apache.ozone.compaction.log.CompactionLogEntry;
+import org.apache.ozone.compaction.log.PopulateCompactionTable;
+import org.apache.ozone.graph.PrintableGraph;
+import org.apache.ozone.rocksdiff.CompactionNode;
+import org.rocksdb.ColumnFamilyDescriptor;
+import org.rocksdb.ColumnFamilyHandle;
+import org.rocksdb.RocksDBException;
+import picocli.CommandLine;
+
+/**
+ * Handler to generate image for current compaction DAG in the OM leader node.
+ * ozone debug om print-log-dag.
+ */
+@CommandLine.Command(
+    name = "print-log-dag",
+    aliases = "pld",
+    description = "Create an image of the current compaction log DAG.")
+public class CompactionLogDagPrinter implements Callable<Void> {
+
+  @CommandLine.Option(names = {"-f", "--file-location"},
+      required = true,
+      description = "Path to location at which image will be downloaded. " +
+          "Should include the image file name with \".png\" extension.")
+  private String imageLocation;
+
+  @CommandLine.Option(names = {"--db"},
+      required = true,
+      scope = CommandLine.ScopeType.INHERIT,
+      description = "Path to OM RocksDB")
+  private String dbPath;
+
+  @CommandLine.Option(names = {"--compaction-log"},
+      required = true,
+      scope = CommandLine.ScopeType.INHERIT,
+      description = "Path to compaction-log directory.")
+  private String compactionLogDir;
+
+  // TODO: Change graphType to enum.
+  @CommandLine.Option(names = {"-t", "--graph-type"},
+      description = "Type of node name to use in the graph image. (optional)\n Accepted values are: \n" +
+          "  FILE_NAME (default) : to use file name as node name in DAG,\n" +
+          "  KEY_SIZE: to show the no. of keys in the file along with file name in the DAG node name,\n" +
+          "  CUMULATIVE_SIZE: to show the cumulative size along with file name in the DAG node name.",
+      defaultValue = "FILE_NAME")
+  private String graphType;
+
+  @Override
+  public Void call() throws Exception {
+    try {
+      System.out.println("tej enter try ");
+      CreateCompactionDag createCompactionDag = new CreateCompactionDag(dbPath, compactionLogDir);
+      System.out.println("tej create compactDag obj");
+      createCompactionDag.pngPrintMutableGraph(imageLocation, PrintableGraph.GraphType.valueOf(graphType));
+      System.out.println("Graph was generated at '" + imageLocation + "'.");
```

Comment:
Please add parent class `AbstractSubcommand` and use `out()` and `err()` instead of `System.out` and `System.err`.